### PR TITLE
[kube-prometheus-stack] bump windows rules and dashboards

### LINKED ISSUE #3817: [kube-prometheus-stack] Outdated windows rules break monitoring
### Describe the bug a clear and concise description of what the bug is.

There was an issue in the upstream windows rules, where starting windows pods could cause record rules like `windows_container_memory_usage` to fail with 
```
many-to-many matching not allowed: matching labels must be unique on one side
```
This was mentioned here https://github.com/kubernetes-monitoring/kubernetes-mixin/issues/612 a while ago, but is actually already fixed in this commit: https://github.com/kubernetes-monitoring/kubernetes-mixin/commit/efd98a492501e8898d75c8ef9c77640e432dd266

We just need to merge these changes into the kube-prometheus-stack. 
There doesn't seem to be an easy way to sync the windows rules and dashboard at the moment, as @lippertmarkus mentioned in the [initial PR](https://github.com/prometheus-community/helm-charts/pull/3083). 

I followed the description in that PR on how to sync the rules and dashboard to provide a PR with the latest version, but it feels like we should have a more robust way to sync windows related rules.

### What's your helm version?

v3.12.3

### What's your kubectl version?

v1.28.2

### Which chart?

kube-prometheus-stack

### What's the chart version?

51.1.0

### What happened?

_No response_

### What you expected to happen?

_No response_

### How to reproduce it?

_No response_

### Enter the changed values of values.yaml?

_No response_

### Enter the command that you execute and failing/misfunctioning.

no specific command. Just need to bump the windows rules/dashboard

### Anything else we need to know?

I already prepared the PR with the updated version.

### PR DESCRIPTION
**Pull request by TheKangaroo**
#### What this PR does / why we need it

bump windows rules and dashboards to latest version

#### Which issue this PR fixes

- fixes #3817

#### Special notes for your reviewer

this bumps the windows rules and dashboards to the latest version.

There are also two minor updates to the node dashboards. I'm not sure if this is incorrect, I can remove these changes if they are incorrect, but the sync script produced these changes.

I created the other changes the same way @lippertmarkus did in the [initial PR](https://github.com/prometheus-community/helm-charts/pull/3083).

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by lippertmarkus** [comment]
@TheKangaroo can you please rebase on the newest main?

**Comment by TheKangaroo** [comment]
I rebased agains latest main

**Comment by TheKangaroo** [comment]
Is there any chance of this being merged soon?
The bug in the old version of alerts forces us to ignore Prometheus rule evaluation errors on our production clusters, which is far from ideal.

File: charts/kube-prometheus-stack/templates/grafana/dashboards-1.14/nodes-darwin.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ Do not change in-place! In order to change this file first read following link:
 https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack/hack
 */ -}}
 {{- $kubeTargetVersion := default .Capabilities.KubeVersion.GitVersion .Values.kubeTargetVersionOverride }}
-{{- if and (or .Values.grafana.enabled .Values.grafana.forceDeployDashboards) (semverCompare ">=1.14.0-0" $kubeTargetVersion) (semverCompare "<9.9.9-9" $kubeTargetVersion) .Values.grafana.defaultDashboardsEnabled .Values.nodeExporter.enabled .Values.nodeExporter.operatingSystems.darwin.enabled }}
```
Comment (QuentinBisson) [inline charts/kube-prometheus-stack/templates/grafana/dashboards-1.14/nodes-darwin.yaml:7]: @TheKangaroo did you change this by hand or using the script?

Comment (QuentinBisson) [reply charts/kube-prometheus-stack/templates/grafana/dashboards-1.14/nodes-darwin.yaml:7]: I think this is coming from https://github.com/prometheus-community/helm-charts/blob/723cfad365d5c55b84162d4a5f11294ea736ca54/charts/kube-prometheus-stack/hack/sync_grafana_dashboards.py#L57. Could you replace the conditions by adding  `.Values.nodeExporter.enabled ` before the operating system conditions and run the sync script one last time?

Comment (TheKangaroo) [reply charts/kube-prometheus-stack/templates/grafana/dashboards-1.14/nodes-darwin.yaml:7]: I think it was changed by the script, but i can revert the change on the non windows rules

Comment (QuentinBisson) [reply charts/kube-prometheus-stack/templates/grafana/dashboards-1.14/nodes-darwin.yaml:7]: I'd rather it's rolled back yes

Comment (TheKangaroo) [reply charts/kube-prometheus-stack/templates/grafana/dashboards-1.14/nodes-darwin.yaml]: I rebuild against latest main (kube-prometheus-stack and kube-prometheus) and added only changes to windows dashboards and rules. should be fine now

**Comment by TheKangaroo** [comment]
@QuentinBisson fixed the changes for non-windows dashboards and rules.
Can you approve?

**Pull request merged**